client/client.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/mikloslorinczi/infra-exec/common"
+	"github.com/pkg/errors"
+)
+
+const envFile = "client.env"
+
+var (
+	nodeName string
+	nodeTags string
+	period   int
+)
+
+func main() { // nolint: gocyclo
+
+	status, err := initConfig()
+	if err != nil {
+		fmt.Println(err)
+		printUsage()
+		os.Exit(status)
+	}
+
+	fmt.Printf("\nInfra Client - %v initialized.\nWith the tags: %v\nPolling Infra Server @ %v every %v seconds\n\n", nodeName, nodeTags, common.APIURL, period)
+
+	for running := true; running; {
+		// Calculate next tick
+		nextTick := time.Now().Add(time.Second * time.Duration(period))
+		// Polling Infra Server for Tasks
+		fmt.Println("Polling Infra Server...")
+		task, ok, err1 := getTaskToExec()
+		if err1 != nil || !ok {
```

Comment:
way too complex, why is there an `ok` and an `error` together?
How should an operation be `ok` when it returns an `error`? 
I quess the variable `ok` is spare here...

Reply:
In case there is no task to execute OK will be false, but this doesn't indicate an error (or should it be an error?)